etcd/etcd.go

```diff
@@ -1439,15 +1443,22 @@ func (obj *EmbdEtcd) nominateCallback(re *RE) error {
 			obj.endpoints[obj.hostname] = addresses // now we have some!
 			// client connects to one of the obj.endpoints servers...
 			log.Printf("Etcd: Addresses are: %s", addresses)
-
+			surls := obj.serverURLs
+			if len(obj.advertiseServerURLs) > 0 {
+				surls = obj.advertiseServerURLs
+			}
 			// XXX: just put this wherever for now so we don't block
 			// nominate self so "member" list is correct for peers to see
-			Nominate(obj, obj.hostname, obj.serverURLs)
+			Nominate(obj, obj.hostname, surls)
 			// XXX: if this fails, where will we retry this part ?
 		}
 
 		// advertise client urls
-		if curls := obj.clientURLs; len(curls) > 0 {
+		curls := obj.clientURLs
+		if len(obj.advertiseClientURLs) > 0 {
```

Comment:
subtle, but I think this len(advClientUrls) should happen nested *within* the earlier len(curls) because if there are no curls specified this will add advertise urls, when it shouldn't.